packages/swarm-components/src/Button.test.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { toMatchImageSnapshot } from 'jest-image-snapshot';
+expect.extend({ toMatchImageSnapshot });
+
+import getScreenRenderer from './screenRenderer';
+
+import Button from './Button';
+
+describe('👀 components are visually the same', function() {
+	let renderer;
+
+	// This is ran when the suite starts up.
+	beforeAll(async () => {
+		renderer = await getScreenRenderer({
+			port: 4000,
+			viewport: { width: 200, height: 100 },
+			staticPath: '../../swarm-styles/dist',
```

Comment:
The way this staticPath is defined is strange to me. This is nice for testing in development (i.e., you make a change in swarm-styles and run `yarn build` and you'll be referencing the latest styles here), however, from a proper dependency management viewpoint I feel as if `swarm-styles` should be a devDependency of `swarm-components` and we reference `main.css` and `global.css` in fashion similar to how we do it in mup-web currently.

Reply:
Agreed. In fact TravisCI might choke as it build builds packages independently.

Reply:
One edge case scenario I can imagine this breaking in - 

You have changes in `swarm-styles` and you run these tests against those changes. You push up the changes to these tests, however for some reason you don't push up changes to `swarm-styles`. Ideally tests will fail and be caught in Travis, but you need to know what you're looking for.

Reply:
@mmcgahan, @hsbacot - can you weight in on this.

Reply:
I think the thing being raised is that these tests are _integration_ tests of `swarm-styles + swarm-components`, not unit tests of `swarm-components`, which is how they are currently configured, so they need to be run in an environment where both packages are guaranteed to exist in their published form.

Two options I can think of:

1. Move these tests to a dedicated 'integration testing' package that depends on `swarm-components + swarm-styles` as dependencies, and imports their 'dist' modules when running tests. I think this would be the most 'formally correct' and robust solution, but it requires some planning to make sure that the publish + test workflow operates as expected
2. Make sure that the command that runs these integration tests is separate from the normal 'unit testing' command, and run the bundle/package command for `swarm-styles` as a prerequesite for running integration tests, e.g.
    ```
    "test:integration": "lerna run build --scope=@meetup/swarm-styles && jest --config jest.int.config.json"
    ```

Good call that these aren't strictly unit tests, though - I think it's easy to miss that.

Extending that thinking, however, is there a reason why component-specific styles are being managed in a package separate from the components? Why not keep tightly-coupled code in the same package? I can see why we would still want a separate `@meetup/swarm-styles` package for 'globals', but we probably want to consider re-organizing some of the component-specific styles to stay closer to their JSX counterparts long term. Separating packages by file type (CSS vs JS) isn't usually the best tactic

Reply:
I feel that we should go with option 2 for now, until we have a better feel for the whole testing setup and avoid premature optimization here.

I'll add a note to Travis CI ticket

Reply:
>Extending that thinking, however, is there a reason why component-specific styles are being managed in a package separate from the components? Why not keep tightly-coupled code in the same package? I can see why we would still want a separate @meetup/swarm-styles package for 'globals', but we probably want to consider re-organizing some of the component-specific styles to stay closer to their JSX counterparts long term. Separating packages by file type (CSS vs JS) isn't usually the best tactic

This @hsbacot would be best to comment on. In general I haven't had any issues with the current setup, up until this ticket where the distinction has made the setup somewhat awkward.

>Move these tests to a dedicated 'integration testing' package that depends on swarm-components + swarm-styles as dependencies, and imports their 'dist' modules when running tests. I think this would be the most 'formally correct' and robust solution, but it requires some planning to make sure that the publish + test workflow operates as expected

I'm in favor of option 1 as an immediate follow up especially if we plan to keep `swarm-components` and `swarm-styles` as separate packages for the time being. (i.e., I don't realistically see us wanting / prioritizing changing this).